gpu_support/add_gpu_support.sh

```diff
@@ -0,0 +1,186 @@
+ver=${ver%.*}
+
+##############################################################################################
+# Check that the CUDA driver version is adequate
+# (
+#  needs to be r450 or r470 which are LTS, other production branches are acceptable but not
+#  recommended, below r450 is not compatible [with an exception we will not explore,see
+#  https://docs.nvidia.com/datacenter/tesla/drivers/#cuda-drivers]
+# )
+# only check first number in case of multiple GPUs
+driver_version=$(nvidia-smi --query-gpu=driver_version --format=csv,noheader | tail -n1)
+driver_version="${driver_version%%.*}"
+# Now check driver_version for compatability
+# Check driver is at least LTS driver R450, see https://docs.nvidia.com/datacenter/tesla/drivers/#cuda-drivers
+if (( $driver_version < 450 )); then
+  echo "Your NVIDIA driver version is too old, please update first.."
+  exit 1
+fi
+
+
+# Check if the CUDA compat libraries are installed and compatible with the target CUDA version
+# if not find the latest version of the compatibility libraries and install them
+
+# get URL to latest CUDA compat libs, exit if URL is invalid
+latest_cuda_compat_url="$(./get_latest_cuda_compatlibs.sh ${os} ${ver} ${eessi_cpu_family})"
+ret=$?
+if [ $ret -ne 0 ]; then
+  echo $latest_cuda_compat_url
+  exit 1
+fi
+
+# Create a general space for our NVIDIA compat drivers
+if [ -w /cvmfs/pilot.eessi-hpc.org/host_injections ]; then
+  mkdir -p /cvmfs/pilot.eessi-hpc.org/host_injections/nvidia
+  cd /cvmfs/pilot.eessi-hpc.org/host_injections/nvidia
+else
+  echo "Cannot write to eessi host_injections space, exiting now..." >&2
+  exit 1
+fi
+
+# Check if we have any version installed by checking for the existence of /cvmfs/pilot.eessi-hpc.org/host_injections/nvidia/latest
+
+driver_cuda_version=$(nvidia-smi  -q --display=COMPUTE | grep CUDA | awk 'NF>1{print $NF}' | sed s/\\.//)
+eessi_cuda_version=$(LD_LIBRARY_PATH=/cvmfs/pilot.eessi-hpc.org/host_injections/nvidia/latest/compat/:$LD_LIBRARY_PATH nvidia-smi  -q --display=COMPUTE | grep CUDA | awk 'NF>1{print $NF}' | sed s/\\.//)
+if [ "$driver_cuda_version" -gt "$eessi_cuda_version" ]; then  echo "You need to update your CUDA compatability libraries"; fi
+
+# Check if our target CUDA is satisfied by what is installed already
+# TODO: Find required CUDA version and see if we need an update
+
+# If not, grab the latest compat library RPM or deb
+# download and unpack in temporary directory, easier cleanup after installation
+mkdir -p tmp
+cd tmp
+compat_file=${latest_cuda_compat_url##*/}
+wget ${latest_cuda_compat_url}
+
+# Unpack it
+# (the requirements here are OS dependent, can we get around that?)
+# (for rpms looks like we can use https://gitweb.gentoo.org/repo/proj/prefix.git/tree/eclass/rpm.eclass?id=d7fc8cf65c536224bace1d22c0cd85a526490a1e)
+# (deb files can be unpacked with ar and tar)
+file_extension=${compat_file##*.}
+if [[ ${file_extension} == "rpm" ]]; then
+  rpm2cpio ${compat_file} | cpio -idmv
```

Comment:
We need to have these supported in the compat layer (and ensure they are being used from there)

Reply:
Can we can do that already in the current pilot @bedroge ?

Reply:
It's not there yet. I can try to look into installing it, together with some other packages/updates (Lmod?). Especially updates are not so straightforward, though.

Reply:
It would be interesting to add the additional packages for now, but I would probably hold off on updates and just wait for the next pilot.

Reply:
> We also probably also need an ENVVAR setting to force display of these modules

In which scenario do you want to force display the modules? The assumption is that if `cudaDir` exists, CUDA itself is also available and only then modules with the `gpu` tag will be shown.

Reply:
I was thinking of login nodes without GPUs, but I guess that scenario is still covered. @kostrzewa was the one who originally raised this so perhaps he see's a reason.

Now that I think about though, what is actually required is an option to skip the various verification steps for that use case, so that you can enable CUDA even from a node without GPUs.

Reply:
Not sure why my comment ended up here...

Shouldn't it be enough to `module use cuda_modules_path` to enable CUDA on nodes without GPUs?

Reply:
What I mean is, what if someone wanted to run `add_gpu_support.sh` from a login node? This would fail because there is no `nvidia-smi` and there is no way to override the checks.

Reply:
@huebner-m I think we still need this boolean right?

Reply:
Yes, the boolean has been added in commit https://github.com/EESSI/software-layer/pull/172/commits/5f82658ee4ea9e5e038ab5d34dacc8b2f8b4b708